tests/test_grid3d/test_grid_xtgformats_io.py

```diff
@@ -17,269 +18,233 @@
 
-def test_grid_export_import_bigcase(tmp_path):
-    """Test io for big case if BIGBOX, and compare xtgf and h5 performance."""
-    grid1 = xtgeo.Grid()
+    fna = benchmark_grid.to_hdf(fname, compression="blosc")
 
-    dimens = (100, 100, 20)
-    if BIGBOX:
-        dimens = (1000, 1000, 100)
+    grd2 = xtgeo.Grid()
 
-    grid1.create_box(dimens)
+    @benchmark
+    def partial_read():
+        grd2.from_hdf(fna, ijkrange=(1, 20, 1, 20, "min", "max"))
 
-    grid2 = xtgeo.Grid()
+    assert grd2.ncol == 20
+    assert grd2.nlay == benchmark_grid.nlay
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigbox.xtgf"
-    grid1.to_xtgf(fname)
-    print("Export bigcase using xtgf: ", xtg.timer(t1))
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigbox.xtgf"
-    grid2.from_xtgf(fname, mmap=True)
-    print("Import bigcase using xtgf and memory mapping: ", xtg.timer(t1))
-    assert grid1._zcornsv.mean() == pytest.approx(grid2._zcornsv.mean())
+@pytest.mark.benchmark(group="import/export")
+def test_benchmark_grid_hdf5_import_blosc_compression(
+    benchmark, tmp_path, benchmark_grid
+):
+    fname = join(tmp_path, "reek_geo_grid.compressed_h5")
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigbox.xtgf"
-    grid2.from_xtgf(fname)
-    print("Import bigcase using xtgf: ", xtg.timer(t1))
+    benchmark_grid._zcornsv += 1.0
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigbox.h5"
-    grid1.to_hdf(fname)
-    print("Export bigcase using hdf5: ", xtg.timer(t1))
+    fna = benchmark_grid.to_hdf(fname, compression="blosc")
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigbox.h5"
-    grid2.from_hdf(fname)
-    print("Import bigcase using h5: ", xtg.timer(t1))
+    grd2 = xtgeo.Grid()
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigboxc.h5"
-    grid1.to_hdf(fname, compression="blosc")
-    print("Export bigcase using hdf5 with compression: ", xtg.timer(t1))
+    @benchmark
+    def read():
+        grd2.from_hdf(fna)
 
-    t1 = xtg.timer()
-    fname = tmp_path / "bigboxc.h5"
-    grid2.from_hdf(fname)
-    print("Import bigcase using h5 with compression: ", xtg.timer(t1))
+    assert_allclose(benchmark_grid._zcornsv, grd2._zcornsv)
+    assert_allclose(benchmark_grid._coordsv, grd2._coordsv)
+    assert_allclose(benchmark_grid._actnumsv, grd2._actnumsv)
 
 
 # ======================================================================================
 # Grid properties:
 
 
-def test_gridprop_export_import_many(tmp_path):
-    """Test exporting etc to xtgcpprop format."""
-    prop1 = xtgeo.GridProperty(REEKPROP1)
+def create_big_prop(testpath):
+    vals = np.zeros(BIGBOX_DIMENSIONS, dtype=np.float32)
+    ncol, nrow, nlay = BIGBOX_DIMENSIONS
+    prp = xtgeo.GridProperty(ncol=ncol, nrow=nrow, nlay=nlay, values=vals)
+    prp.values[0, 0, 0:3] = 33
+    prp.values[1, 0, 0:3] = 66
+    prp.values[1, 1, 0:3] = 44
+    return prp
 
-    print(prop1.values1d)
 
-    nrange = 50
+def create_small_prop(testpath):
```

Comment:
and testpath not needed here either?

Reply:
Not needed, but its to keep the same signature as the other creators. I could fix it with a lambda, but it seemed more verbose than the current approach.